cohorts/2024/02-workflow-orchestration/homework.md

```diff
@@ -0,0 +1,39 @@
+## Week 2 Homework
+
+For the homework, we'll be working with the _green_ taxi dataset located here:
+
+`https://github.com/DataTalksClub/nyc-tlc-data/releases/download/green`
+
+You can find the Homework overview video [here](https://youtu.be/_Fey4fTTZ5g).
+
+### Assignment
+
+The following are steps to constructing an ETL pipeline using Mage that were covered in the course. 
+
+- Create a new pipeline, call it `green_taxi_etl`
+- Add a data loader block and use Pandas to read data for the final quarter of 2020 (months `10`, `11`, `12`).
+  - You can use the same datatypes and date parsing methods shown in the course.
+  - `BONUS`: load the final three months using a for loop and `pd.concat`
+- Add a transformer block and perform the following:
+  - Remove rows where the passenger count is equal to 0 _or_ the trip distance is equal to zero.
+  - Create a new column `lpep_pickup_date` by converting `lpep_pickup_datetime` to a date.
+  - Rename columns in Camel Case to Snake Case, e.g. `VendorID` to `vendor_id`.
+  - Add three assertions:
+    - `vendor_id` is one of `1` or `2`
+    - `passenger_count` is greater than 0
+    - `trip_distance` is greater than 0
+- Using a Postgres data exporter (SQL or Python), write the dataset to a table called `green_taxi` in a schema `mage`. Replace the table if it already exists.
+  - `BONUS`: Write your data as Parquet files to a bucket in GCP, partioned by `lpep_pickup_date`. Use the `pyarrow` library!
+- Schedule your pipeline to run daily at 5AM UTC.
+- Join the [Mage Slack](https://mage.ai/chat) channel and say hi 👋
```

Comment:
That shouldn't be a part of the homework, people will join the community because you're awesome, not because it's a part of the homework